rta/src/main/java/com/gluonhq/richtextarea/model/UnitBuffer.java

```diff
@@ -52,7 +52,7 @@
 public class UnitBuffer {
 
     private static final Pattern BLOCK_PATTERN = Pattern.compile(
-            TextBuffer.ZERO_WIDTH_NO_BREAK_SPACE_TEXT + "(@|#)([\\p{L}|\\p{N}|\\s]*)" + TextBuffer.ZERO_WIDTH_NO_BREAK_SPACE_TEXT,
+            TextBuffer.ZERO_WIDTH_NO_BREAK_SPACE_TEXT + "([@#])([\\p{L}\\p{N}\\p{P}\\s]*)" + TextBuffer.ZERO_WIDTH_NO_BREAK_SPACE_TEXT,
             Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.DOTALL);
```

Comment:
It would be nice to have some documentation on this block describing the pattern or at least the regex.

Reply:
done